### PR TITLE
feat: regenerate lsp templates after LvimUpdate

### PR DESCRIPTION
**Pull request by LostNeophyte**

### CODE REVIEW

**Comment by cpea2506** [comment]
https://github.com/LunarVim/LunarVim/blob/0a0cf29c8aa1fafb09dac1a36e30dec642ed24a9/lua/lvim/utils/hooks.lua#L59

This line already does the template regeneration for us.

**Comment by LostNeophyte** [comment]
> https://github.com/LunarVim/LunarVim/blob/0a0cf29c8aa1fafb09dac1a36e30dec642ed24a9/lua/lvim/utils/hooks.lua#L59
> 
> This line already does the template regeneration for us.

but it does it before updating plugins

**Comment by cpea2506** [comment]
> > https://github.com/LunarVim/LunarVim/blob/0a0cf29c8aa1fafb09dac1a36e30dec642ed24a9/lua/lvim/utils/hooks.lua#L59
> > 
> > This line already does the template regeneration for us.
> 
> but it does it before updating plugins

I couldn't see the difference here 😅. I have two options:
1) I think we can re-order them: Plugin update -> Reset cache
2) Remove this line instead
https://github.com/LunarVim/LunarVim/blob/0a0cf29c8aa1fafb09dac1a36e30dec642ed24a9/lua/lvim/utils/hooks.lua#L39

**Comment by LostNeophyte** [comment]
>     1. Re-order them: Plugin update -> Reset cache
clearing cache before updating may have benefits, if we change `enabled` flag of some plugins I'm not sure if lazy will pick this up without clearing cache

>     2. Remove this line and let `post-hook` handle it
>        https://github.com/LunarVim/LunarVim/blob/0a0cf29c8aa1fafb09dac1a36e30dec642ed24a9/lua/lvim/utils/hooks.lua#L39

we can't remove that line, it's used in LvimCacheReset

**Comment by cpea2506** [comment]
> > ```
> > 1. Re-order them: Plugin update -> Reset cache
> > ```
> 
> 
> clearing cache before updating may have benefits, if we change `enabled` flag of some plugins I'm not sure if lazy will pick this up without clearing cache
> 
> 
> > ```
> > 2. Remove this line and let `post-hook` handle it
> >    https://github.com/LunarVim/LunarVim/blob/0a0cf29c8aa1fafb09dac1a36e30dec642ed24a9/lua/lvim/utils/hooks.lua#L39
> > ```
> 
> 
> we can't remove that line, it's used in LvimCacheReset

I almost forgot it was used in a command. 🥲

File: lua/lvim/utils/hooks.lua [hunk 1]
```diff
@@ -61,6 +61,7 @@ function M.run_post_update()
   Log:debug "Syncing core plugins"
   plugin_loader.reload { reload "lvim.plugins", lvim.plugins }
   plugin_loader.sync_core_plugins()
+  require("lvim.lsp.templates").generate_templates()
```
Comment (kylo252) [inline lua/lvim/utils/hooks.lua:64]:
```suggestion
  M.reset_cache() -- force cache clear and templates regen once more  
```

**Pull request merged**